### PR TITLE
refactor: update `blas/ext/base/dnansumkbn2` to follow current project conventions

### LINKED ISSUE #1485: [RFC]: refactor `blas/ext/base/dnansumkbn2` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dnansumkbn2`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dnansumkbn2) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by kumarsuraj212003**
Resolves #1485 .

## Description

> What is the purpose of this pull request?

This pull request:

update `blas/ext/base/dnansumkbn2` to follow current project conventions

## Related Issues
#788, #1152

> Does this pull request have any related issues?

This pull request:

-   progresses #1152
-   fixes #1485

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/src/addon.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+* Receives JavaScript callback invocation data.
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value node_dnansumkbn2( napi_env env, napi_callback_info info ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/src/addon.c]:
```suggestion
static napi_value addon( napi_env env, napi_callback_info info ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/manifest.json [hunk 2]
```diff
@@ -1,42 +1,76 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/napi/export",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dnansumkbn2/manifest.json:42]: Still need `@stdlib/math/base/assert/is-nan` as used in `dnansumkbn2.c`.

```suggestion
        "@stdlib/math/base/assert/is-nan",
        "@stdlib/napi/export",
```

**Comment by Planeshifter** [comment]
@kumarsuraj212003 Thanks for your PR! There were a few errors in the `manifest.json`, so please to make sure going forward to test out your changes with the `make` commands for running C benchmarks, examples, and tests.

Tested everything locally and will merge shortly.

**Comment by kumarsuraj212003** [comment]
@Planeshifter Thank you for the suggestion; I will make sure to run tests locally. Your advice is appreciated.

**Pull request merged**